src/ansi-c/ansi_c_parser.cpp

```diff
@@ -175,3 +175,55 @@ ansi_c_id_classt ansi_c_parsert::get_class(const typet &type)
 
   return ansi_c_id_classt::ANSI_C_SYMBOL;
 }
+
+bool ansi_c_parsert::pragma_cprover_empty()
+{
+  return pragma_cprover_stack.empty();
+}
+
+void ansi_c_parsert::pragma_cprover_push()
+{
+  std::map<irep_idt, bool> empty;
+  pragma_cprover_stack.push_back(std::move(empty));
```

Comment:
I'm wondering whether `pragma_cprover_stack.emplace_back({});` would actually work (and be more concise)?